examples/ex29.cpp

```diff
@@ -0,0 +1,375 @@
+//               coefficient.
+//
+//               This example demonstrates the use of finite element
+//               integrators on 2D domains with 3D coefficients.
+//
+//               We recommend viewing example 1 before viewing this example.
+
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace std;
+using namespace mfem;
+
+Mesh * GetMesh(int type);
+
+void trans(const Vector &x, Vector &r);
+
+void sigmaFunc(const Vector &x, DenseMatrix &s);
+
+double uExact(const Vector &x)
+{
+   return (0.25 * (2.0 + x[0]) - x[2]) * (x[2] + 0.25 * (2.0 + x[0]));
+}
+
+void duExact(const Vector &x, Vector &du)
+{
+   du.SetSize(3);
+   du[0] = 0.125 * (2.0 + x[0]) * x[1] * x[1];
+   du[1] = -0.125 * (2.0 + x[0]) * x[0] * x[1];
+   du[2] = -2.0 * x[2];
+}
+
+void fluxExact(const Vector &x, Vector &f)
+{
+   f.SetSize(3);
+
+   DenseMatrix s(3);
+   sigmaFunc(x, s);
+
+   Vector du(3);
+   duExact(x, du);
+
+   s.Mult(du, f);
+   f *= -1.0;
+}
+
+int main(int argc, char *argv[])
+{
+   // 1. Parse command-line options.
+   int order = 3;
+   int mesh_type = 4; // Default to Quadrilateral mesh
+   int ref_levels = 0;
+   bool static_cond = false;
+   bool visualization = true;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_type, "-mt", "--mesh-type",
+                  "Mesh type: 3 - Triangular, 4 - Quadrilateral.");
+   args.AddOption(&ref_levels, "-r", "--refine",
+                  "Number of times to refine the mesh uniformly in serial.");
+   args.AddOption(&order, "-o", "--order",
+                  "Finite element order (polynomial degree) or -1 for"
+                  " isoparametric space.");
+   args.AddOption(&static_cond, "-sc", "--static-condensation", "-no-sc",
+                  "--no-static-condensation", "Enable static condensation.");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.Parse();
+   if (!args.Good())
+   {
+      args.PrintUsage(cout);
+      return 1;
+   }
+   args.PrintOptions(cout);
+
+   // 2. Construct a quadrilateral or triangular mesh with the topology of a
+   //    cylindrical surface.
+   Mesh *mesh = GetMesh(mesh_type);
+   int dim = mesh->Dimension();
+
+   // 3. Refine the mesh to increase the resolution. In this example we do
+   //    'ref_levels' of uniform refinement.
+   for (int l = 0; l < ref_levels; l++)
+   {
+      mesh->UniformRefinement();
+   }
+
+   // 4. Transform the mesh so that it has a more interesting geometry.
+   mesh->SetCurvature(3);
+   mesh->Transform(trans);
+
+   // 5. Define a finite element space on the mesh. Here we use continuous
+   //    Lagrange finite elements of the specified order. If order < 1, we
+   //    instead use an isoparametric/isogeometric space.
+   FiniteElementCollection *fec;
+   bool delete_fec;
+   if (order > 0)
```

Comment:
As I mentioned before, we probably do not want the "isoparametric" option which is what we handle here with this `if () { ... } else { ... }` statement.